### DISCUSSION TITLE
Sozi Presentation as an app

### DISCUSSION THREAD

**Discussion by waelkasim** [senshu/Sozi#542] [Ideas]
Sozi is a great presentation tool, I use it and particularly love the fact that I can work on the SVG file on Inkscape and update the presentation on Sozi instantly. This strength can also be a weakness; as most of the issues I run into with Sozi are either Inkscape-related or, most importantly, browser related. I can usually work around the Inkscape related issues, but I'm left at the mercy of the end-user's browser (since I mostly send my presentations to clients not present them at their premises on my machine).

I want to package my presentations as desktop apps to eliminate any possibility of jank and stuttering at the end-user's end and make sure that my presentations run as smoothly as they do inside Sozi.

My first thought was to pack my presentation with CEF-Chromium Embedded Framework so that when the users opens up my presentation/app it runs from the embedded browser totally independent of the user's browser(s).

But I know that Sozi is an Electron app which uses the chromium engine for the UI instead of CEF, which was an initial design decision that was abandoned for the reasons in this blog: https://www.electronjs.org/blog/electron-internals-building-chromium-as-a-library
and so this might not be the most efficient way.

My question (perhaps feature request) is: can Sozi embed a version of itself into the presentation - so that instead of running the presentation on an external browser, it actually runs on the same engine that it was created with? yet of course without the UI?

Thank you and keep up the good work.

**Comment by ejvindh** [comment]
I think that in some usercases this might be benefitial. In other cases, however, the problem with the idea would be that instead of being at the mercy of browser-choice you will become dependent on which OS, the user runs. In current settings, you can make your presentation, and then ask the user to present with (e.g.) Chrome or Firefox. Then they will be able to run it on whatever device they use. If you embedded the browser, you would probably have to specify which OS it should be run in.

**Reply by waelkasim** [reply]
Thank you, ejvindh, for your response. I agree with you. But, asking the user to present with (e.g.) Chrome or Firefox, doesn't differ much from asking them about their OS.

**Comment by aumouvantsillage** [comment]
@waelkasim Sozi is designed with the assumption that there is a fairly recent web browser installed on every computer, so that anybody could play a presentation without installing anything. The idea of packaging a web browser engine with the presentation had never occurred to me :) And until now, I was under the impression that the differences between web browsers were either negligible or easy to work around. But I see that you have a different experience.

Even if the presentation editor is based on Electron, I think that CEF is still relevant for playing Sozi presentations. The player only needs a web browser engine and does not use any Electron-specific features. Ideally, a presentation should be packaged with the same version of Chromium that Electron uses.

> My question (perhaps feature request) is: can Sozi embed a version of itself into the presentation

Converting a Sozi presentation to a standalone application is totally feasible.
However, it will make each presentation file much heavier (and platform-dependent, as @ejvindh pointed out) and I wonder whether it would not be more efficient to distribute a separate Chromium-based lightweight web browser that would be rebranded as the "Sozi viewer" app.

**Reply by waelkasim** [reply]
Thank you, Guillaume, for your response.

> I was under the impression that the differences between web browsers were either negligible or easy to work around. But I see that you have a different experience.

Oh they differ greatly in my experience; See, my presentations are fairly heavy with SVG and PNG images and Arabic text. They run beautifully on Sozi's editor but they stutter in 3 specific slides on Chrome and down right unreadable on Firefox - both are the latest stable versions, mind you.

> Ideally, a presentation should be packaged with the same version of Chromium that Electron uses.

Yes, that's exactly what I'm aiming for. And, yes, it would probably "be more efficient to distribute a separate Chromium-based lightweight web browser that would be rebranded as the "Sozi viewer" app.".. more efficient & lighter weight then this example: https://github.com/nwjs/nw.js which requires you to move the presentation folder to nw.js and ship it for more than a 100mb zipped!

What I'm looking for is an experience where the user receives my presentation on a CD or USB, finds an icon, clicks it and voila: the presentation is running its own Chromium/CEF exactly as it runs in Sozi. This could be an OS-specific solution as @ejvindh suggested but I'm fine with that.
https://www.todesktop.com/ This service does that for a premium.

What we're talking about here is called https://en.wikipedia.org/wiki/Site-specific_browser

In my mind the most efficient way to do this is utilize Sozi's built-in Chromium to export presentation as cross-platform desktop app.. kinda like PWA (Progressive Web Apps) but with a full installation on the user's desktop. What do you guys think?
